tests/test_views.py

```diff
@@ -4583,5 +4529,269 @@ def test_confirm_user_signal_called_when_user_confirms_email(self):
 
 
+class TestDraftRegistrationViews(OsfTestCase):
+
+    def setUp(self, *args, **kwargs):
+        super(TestDraftRegistrationViews, self).setUp(*args, **kwargs)
+
+        self.user = AuthUserFactory()
+        self.auth = self.user.auth
+        self.node = ProjectFactory(creator=self.user)
+
+        MetaSchema.remove()
+        ensure_schemas()
+        self.meta_schema = MetaSchema.find_one(
+            Q('name', 'eq', 'Open-Ended Registration') &
+            Q('schema_version', 'eq', 1)
+        )
+        self.draft = DraftRegistrationFactory(
+            initiator=self.user,
+            branched_from=self.node,
+            registration_schema=self.meta_schema,
+            registration_metadata={
+                'summary': {'value': 'Some airy'}
+            }
+        )
+        self.draft.save()
+
+        current_month = datetime.datetime.now().strftime("%B")
+        current_year = datetime.datetime.now().strftime("%Y")
+
+        valid_date = datetime.datetime.now() + datetime.timedelta(days=180)
+        self.valid_embargo_payload = json.dumps({
+            u'embargoEndDate': unicode(valid_date.strftime('%a, %d, %B %Y %H:%M:%S')) + u' GMT',
+            u'registrationChoice': 'embargo',
+            u'summary': unicode(fake.sentence())
+        })
+        self.invalid_embargo_date_payload = json.dumps({
+            u'embargoEndDate': u"Thu, 01 {month} {year} 05:00:00 GMT".format(
+                month=current_month,
+                year=str(int(current_year) - 1)
+            ),
+            u'registrationChoice': 'embargo',
+            u'summary': unicode(fake.sentence())
+        })
+
+    def tearDown(self, *args, **kwargs):
+        super(TestDraftRegistrationViews, self).tearDown(*args, **kwargs)
+        DraftRegistration.remove()
+
+    @unittest.skip('TODO: test when we support draft review')
+    def test_submit_draft_for_review(self):
+        pass
+
+    @mock.patch('website.project.model.DraftRegistration.register', autospec=True)
+    def test_register_draft_registration(self, mock_register_draft):
+
+        url = self.node.api_url_for('register_draft_registration', draft_id=self.draft._id)
+        res = self.app.post_json(url, {
+            'registrationChoice': 'Make registration public immediately'
+        }, auth=self.auth)
+
+        assert_equal(res.status_code, http.ACCEPTED)
+        assert_equal(mock_register_draft.call_args[0][0]._id, self.draft._id)
+
+    @mock.patch('framework.tasks.handlers.enqueue_task')
+    def test_register_template_make_public_creates_pending_registration(self, mock_enquque):
+        url = self.node.api_url_for('register_draft_registration', draft_id=self.draft._id)
+        res = self.app.post_json(url, {'registrationChoice': 'immediate'}, auth=self.auth)
+
+        assert_equal(res.status_code, http.ACCEPTED)
+        self.node.reload()
+        # Most recent node is a registration
+        reg = Node.load(self.node.node__registrations[-1])
```

Comment:
Is there any situation where this registration will not exist yet by the time this line is reached? (eg if registration is created in a background task)